### DISCUSSION TITLE
Restoring from S3 disk

### DISCUSSION THREAD

**Discussion by jeeva-travtech** [ClickHouse/ClickHouse#61283] [Q&A]
I have configured Clickhouse cluster storage_policy to move the rows of a table to S3 based on a TTL 
these are my configurations 

```
<clickhouse>
  <s3>
    <use_environment_credentials>true</use_environment_credentials>
  </s3>
  <storage_configuration>
    <disks>
      <s3_disk>
        <type>s3</type>
        <endpoint>https://clickhouse************.s3.**********.amazonaws.com/</endpoint>
        <metadata_path>/var/lib/clickhouse/disks/s3_disk/</metadata_path>
      </s3_disk>
    </disks>
    <policies>
      <tiered>
        <volumes>
          <default>
            <disk>default</disk>
          </default>
          <s3_disk>
            <disk>s3</disk>
          </s3_disk>
        </volumes>
      </tiered>
    </policies>
  </storage_configuration>
</clickhouse>
```
With this configuration, Clickhouse was able to write to S3 after TTL has expired as shown below

![image](https://github.com/ClickHouse/ClickHouse/assets/160855244/54f9db17-3d4c-4ad0-a664-d1d362bc442b)

How will I be able to 'Load' / 'Restore' it back to a new Clickhouse instance again for analysis purposes ?

**Comment by yiyang-shao** [comment]
Hi! Does this: https://clickhouse.com/docs/en/operations/backup#backup-to-a-local-disk suit your needs?

**Comment by jeeva-travtech** [comment]
Hi, thanks for the reply , however, my use case is not to take backups since the data is there in S3 already (based on a TTL the records are moved to S3 as per the above configs) , I would just like to use the S3 data as a disk and able to see the databases, tables that were moved based on the TTL

**Reply by yiyang-shao** [reply]
Sorry, im not sure if I understand your use case correctly. Do you aim to create a new clickhouse instance with the data stored on the s3 of the existing clickhouse instance only?

**Reply by jeeva-travtech** [reply]
yes, that's my use case, my understanding is that whatever data is there in S3 , if I spawn a new Clickhouse instance and configure to use S3 as a disk it should automatically work.

But looks like it wont because the disk just has the 'raw data' might not have others such as metadata and stuff, so just asking if my train of thought is correct .

If it is not possible , then what can be done to achieve it ?

**Reply by yiyang-shao** [reply]
> But looks like it wont because the disk just has the 'raw data' might not have others such as metadata and stuff, so just asking if my train of thought is correct .

This is correct. The metadata still resides in clickhouse-server local disk and keeper. So the raw data part file on s3 cannot be directly consumed by another Clickhouse instance.

> If it is not possible , then what can be done to achieve it ?

The possible approach will be an export&import to the new clickhouse instance. However, may I know more details that what kind of analysis of the cold data on s3 you expect to conduct on a separate clickhouse instance? I will see if I have any good suggestions to the specific use cases.

**Reply by jeeva-travtech** [reply]
Hmmm, by export and import I would assume a backup and restore process.
My assumption was just after I set the configuration with S3 as a disk ,'it just works' , but I was wrong.

However, my use case is such that , the data in S3 will be loaded only during the case of debugging issues and to be used security incidents, which are very important as we need to set up a process so that we don't end up spending time to set things up.

Would be really great if there is any other approach apart from backup/restore.

**Reply by yiyang-shao** [reply]
Hi I personally have not tried with a multi-tier setup, but you may attempt to use s3 table engine combined with clickhouse native format.
* https://clickhouse.com/docs/en/engines/table-engines/integrations/s3
* https://clickhouse.com/docs/en/integrations/data-formats/binary-native

**Reply by jeeva-travtech** [reply]
Sorry for the late reply , I am really not sure how to go about with the above approach, S3 engine needs a path which is something compressed but we don't have those paths when S3 is used as disk, not sure how to go about it

**Comment by saisumit** [comment]
@jeeva-travtech  @yiyang-shao  Any luck on this?
